modules/partition-replicator/src/integrationTest/java/org/apache/ignite/internal/partition/replicator/ItReplicaLifecycleTest.java

```diff
@@ -656,6 +660,41 @@ public void testScanCloseReplicaRequest() throws Exception {
         assertDoesNotThrow(tx::commit);
     }
 
+    @Test
+    public void testReplicaSafeTimeSyncRequest() throws Exception {
+        // Prepare a single node cluster.
+        startCluster(2);
+        Node node0 = getNode(0);
+        List<Set<Assignment>> assignments = PartitionDistributionUtils.calculateAssignments(
+                cluster.stream().map(n -> n.name).collect(toList()), 1, 1);
+
+        List<TokenizedAssignments> tokenizedAssignments = assignments.stream()
+                .map(a -> new TokenizedAssignmentsImpl(a, Integer.MIN_VALUE))
+                .collect(toList());
+
+        placementDriver.setPrimary(node0.clusterService.topologyService().localMember());
+        placementDriver.setAssignments(tokenizedAssignments);
+
+        // Prepare a zone.
+        String zoneName = "test_zone";
+        createZone(node0, zoneName, 1, 2);
+        int zoneId = DistributionZonesTestUtil.getZoneId(node0.catalogManager, zoneName, node0.hybridClock.nowLong());
+        int partId = 0;
+
+        // Create a table to work with.
+        createTable(node0, zoneName, "test_table");
+
+        HybridTimestamp node0safeTimeBefore = node0.currentSafeTimeForZonePartition(zoneId, partId);
```

Comment:
currentSafeTimeForZonePartition may fail with` Missing resources for zone` if you will call 
```
    ZonePartitionResources getZonePartitionResources(ZonePartitionId zonePartitionId) {
        ZoneResources zoneResources = resourcesByZoneId.get(zonePartitionId.zoneId());
```
fast enough.

Reply:
Now I'm asserting zone and table creation explicitly